compiler/ast/ast_query.nim

```diff
@@ -662,4 +662,50 @@ proc skipAddr*(n: PNode): PNode {.inline.} =
 
 proc isNewStyleConcept*(n: PNode): bool {.inline.} =
   assert n.kind == nkTypeClassTy
-  result = n[0].kind == nkEmpty
\ No newline at end of file
+  result = n[0].kind == nkEmpty
+
+
+type
+  NamedNodePos* = enum
```

Comment:
```suggestion
  NamedNodePos* = enum
    ## nodes that have a name like proc definitions, identifier definitions in let sections, etc
```

I think that captures the intention?

Reply:
A pragma pos would be nice, the index varies per kind but it'd be super helpful.

Reply:
Pragma position for different entry types, like procedure, type declaration etc?

Reply:
Yup